interactions.js

```diff
@@ -9,4 +9,31 @@ document.getElementById("mySubmit").onclick = function(){
 document.getElementById("ideaSubmit").onclick = function(){
     idea = document.getElementById("ideaInput").value;
     console.log(idea);
+}
+
+ // Initialize modal
+document.addEventListener('DOMContentLoaded', function() {
+    var elems = document.querySelectorAll('.modal');
+    var instances = M.Modal.init(elems);
+});
+
+// Function to display selected pastry
+function displayPastry() {
+    const selectedPastry = document.querySelector('input[name="pastry"]:checked').value;
+    document.getElementById('pastry-response').innerText = `You selected: ${selectedPastry}`;
+}
+
+// Function to handle email subscription form submission
+function submitEmail(event) {
+    event.preventDefault();
+    const email = document.getElementById('email').value;
+    document.getElementById('email-response').innerText = `Thank you for subscribing! Updates will be sent to: ${email}`;
+    document.getElementById('email').value = ''; // Clear the input
+
+
+    // Display modal
+    const modalText = document.getElementById('modal-content-text');
+    modalText.innerText = `Thank you for subscribing! Updates will be sent to: ${email}`;
+    const modal = M.Modal.getInstance(document.getElementById('modal1'));
+    modal.open();
```

Comment:
@FoxfordOH This code isn't working either, you'll also want to move it to the materialize page.